apps/home/routes.py

```diff
@@ -393,6 +393,157 @@ def view_labs(lab_id=None):
 
 
+@blueprint.route('/groups/', defaults={'group_id': None})
+@blueprint.route('/groups/<int:group_id>')
+@login_required
+def groups(group_id):
+    if current_user.category == "user":
+        return render_template('pages/waiting_approval.html')
+
+    if group_id:
+        group = Groups.query.get_or_404(group_id)
+        return render_template('pages/group_info.html', group=group)
+
+    groups = []
+    try:
+        user_groups = k8s.get_groups_by_user(current_user.uid)  
+    except Exception as exc:
+        err = traceback.format_exc().replace("\n", ", ")
+        current_app.logger.error(f"Error getting groups: {exc} -- {err}")
+        return render_template("pages/error.html", title="Error getting groups", msg="Failed to obtain groups. Check logs for more information.")
+
+    registered_groups = {group.id: group.groupname for group in Groups.query.all()}
+    registered_users = {user.uid: user for user in Users.query.all()}
+
+    for group_id, user_uid in user_groups:
+        user = registered_users.get(user_uid)
+        if not user:
+            current_app.logger.warning(f"Inconsistency found on group: owner user not found in database {user_uid=} ({group_id=})")
+            continue
+        group = Groups.query.filter_by(id=group_id).first()
+        if not group:
+            current_app.logger.warning(f"Inconsistency found on group: group not found for {group_id=}")
+            continue
+
+        group_dict = {
+            "groupname": registered_groups.get(group_id, f"Unknown Group {group_id}"),
+            "group_id": group_id,
+            "user": f"{user.given_name} {user.family_name} ({user.email or 'NO-EMAIL'})",
+            "members": [],
+        }
+
+        created = None
+        member_ids = [group.owner_id, group.assistant_id, group.member_id]
+        for member_id in member_ids:
+            member = Users.query.get(member_id)
+            if member:
+                created = member.created  
+                group_dict["members"].append({
+                    "name": f"{member.given_name} {member.family_name}",
+                    "role": "Owner" if member_id == group.owner_id else "Assistant" if member_id == group.assistant_id else "Member",
+                    "joined": created.strftime('%Y-%m-%d %H:%M:%S') if created else "--",
+                    "status": "Active" if member.active else "Inactive",
+                })
+
+        if created:
+            group_dict["created"] = created.strftime('%Y-%m-%d %H:%M:%S')
+        else:
+            group_dict["created"] = "--"
+        groups.append(group_dict)
+
+    return render_template("pages/group_info.html", segment="groups", groups=groups)
+
+
+@blueprint.route('/group/', defaults={'group_id': None}, methods=["GET"])
+@blueprint.route('/group/<int:group_id>', methods=["GET"])
+@login_required
+def view_group(group_id):
+
```

Comment:
Please start by checking if the current_user.category == "user" and then return the not authorized message!